### PR TITLE
Don't crash if can't find sqlstat1 table for fdb

### PR DESCRIPTION
**Pull request by chands10**
This happens when querying local_fdb table without sqlite_stat1.
With this patch results get returned, but db prints out `[ERROR] Failed local match fdb testdb2 class 5 local 1, asked for class 5 local 0`

To help us review your pull request, please consider providing an overview of the following:
* What is the type of the change (bug fix, feature, documentation and etc.) ?
* What are the current behavior and expected behavior, if this is a bugfix ?
* What are the steps required to reproduce the bug, if this is a bugfix ?
* What is the current behavior and new behavior, if this is a feature change or enhancement ?
* [Optional] Why is the new behavior better than the current behavior, if this is a feature change ?

### CODE REVIEW

**Comment by chands10** [comment]
Will add test

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Error &#x26A0;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>4/487 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>systable_locking<br>sc_truncate<br>queuedb_rollover_noroll1_generated<br>sys_cmd_access</p>

File: db/fdb_fend.c
Comment (riverszhang89) [inline db/fdb_fend.c:696]: I'd avoid using hardcoded error codes (-1 is generic enough, though). It also appears that we do not handle the error codes that differently. Can we move the logmsg calls for printing errors into this function, and return -1 for both error cases? For instance,

```
if (!stat_tbl) {
    logmsg(LOGMSG_ERROR, "%s: OOM stat1 for %p\n", __func__, tbl);
    return -1;
}

// ...

if (!stat_ent) {
    logmsg(LOGMSG_ERROR, "%s: Cannot find stat1 for %p\n", __func__, tbl);
    return -1;
}

```

Comment (chands10) [reply db/fdb_fend.c:696]: That makes sense, will do!

**Review by riverszhang89 — APPROVED** [review]

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Error &#x26A0;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>7/487 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>phys_rep_tiered<br>queuedb_rollover_noroll1_generated<br>queuedb_rollover<br>limitsortingtbl<br>sys_cmd_access<br>transchunk_manyins<br>fdb_push</p>

**Comment by chands10** [comment]
Test fails with `[select * from LOCAL_remdb2fdbpush70499.t order by id] failed with rc -3 mismatching class for db "remdb2fdbpush70499"`

**Comment by dorinhogea** [comment]
> Test fails with `[select * from LOCAL_remdb2fdbpush70499.t order by id] failed with rc -3 mismatching class for db "remdb2fdbpush70499"`
> 
> But also doesn't seg fault anymore :)

This is fixed in https://github.com/bloomberg/comdb2/pull/3995

**Comment by chands10** [comment]
Sounds good! I'll wait until that patch is merged

**Review by riverszhang89 — APPROVED** [review]

**Comment by chands10** [comment]
fdb_push test is still failing, going to wait for https://github.com/bloomberg/comdb2/pull/3999 to be merged before this

File: tests/fdb_push.test/test_fdb_push.sh
Comment (chands10) [inline tests/fdb_push.test/test_fdb_push.sh:92]: This line still fails with mismatching class, so just make sure this line doesn't seg fault for now

File: tests/fdb_push.test/test_fdb_push.sh
Comment (chands10) [inline tests/fdb_push.test/test_fdb_push.sh:92]: This line still fails with mismatching class, so just make sure this line doesn't seg fault for now since that is the point of this patch, don't add to output

**Review by riverszhang89 — APPROVED** [review]

**Comment by chands10** [comment]
Thank you Rivers x3!

**Pull request merged**
